spec/exporters/dhis2_exporter_spec.rb

```diff
@@ -0,0 +1,181 @@
+      dummy_period = 'dummy2020'
+      expected_data_values = facility_identifiers.map { |facility_identifier|
+        periods.product(data_elements_map.values).map { |_period, data_element_id|
+          {
+            data_element: data_element_id,
+            org_unit: facility_identifier.identifier,
+            period: dummy_period,
+            value: dummy_value
+          }
+        }
+      }
+      exporter = described_class.new(
+        facility_identifiers: facility_identifiers,
+        periods: periods,
+        data_elements_map: data_elements_map
+      )
+
+      allow(exporter).to receive(:reporting_period).and_return(dummy_period)
+
+      allow(exporter).to receive(:send_data_to_dhis2)
+      expect(exporter).to receive(:send_data_to_dhis2).with(expected_data_values.first)
+      expect(exporter).to receive(:send_data_to_dhis2).with(expected_data_values.second)
+
+      exporter.export do |_facility_identifier, _period|
+        {
+          data_elements_map.keys.first => dummy_value,
+          data_elements_map.keys.second => dummy_value
+        }
+      end
+    end
+  end
+
+  describe '#export_disaggregated' do
+    it 'sends disaggregated data element values for all facilities, given facility identifiers, periods and data elements' do
+      dummy_value = 0
+      dummy_period = 'dummy2020'
+      expected_data_values = facility_identifiers.map do |facility_identifier|
+        periods.product(data_elements_map.values).map do |_period, data_element_id|
+          category_option_combo_ids.map do |_combo, id|
+            {
+              data_element: data_element_id,
+              org_unit: facility_identifier.identifier,
+              category_option_combo: id,
+              period: dummy_period,
+              value: dummy_value
+            }
+          end
+        end.flatten
+      end
+      exporter = described_class.new(
+        facility_identifiers: facility_identifiers,
+        periods: periods,
+        data_elements_map: data_elements_map,
+        category_option_combo_ids: category_option_combo_ids
+      )
+
+      allow(exporter).to receive(:reporting_period).and_return(dummy_period)
+      # TODO: check disagg call is happening
+      allow(exporter).to receive(:send_data_to_dhis2)
+      expect(exporter).to receive(:send_data_to_dhis2).with(expected_data_values.first)
+      expect(exporter).to receive(:send_data_to_dhis2).with(expected_data_values.second)
+
+      exporter.export_disaggregated do |_facility_identifier, _period|
+        {
+          data_elements_map.keys.first => category_option_combo_ids.transform_values {|_value| dummy_value},
+          data_elements_map.keys.second => category_option_combo_ids.transform_values {|_value| dummy_value}
+        }
+      end
+    end
+  end
+
+  describe '#disaggregate_data_values' do
+    it 'should return a list of disaggregated data values for each category-option combo for the given data element and period' do
+      dummy_value = 5
```

Comment:
Given tests have dummy data in general, these names don't need to be prefixed with `dummy`. Can just call them `value` and `period`.